pkg/activator/net/helpers.go

```diff
@@ -37,6 +37,12 @@ func EndpointsToDests(endpoints *corev1.Endpoints, portName string) []string {
 					// Prefer IP as we can avoid a DNS lookup this way.
 					ret = append(ret, net.JoinHostPort(addr.IP, portStr))
 				}
+				// Append not ready addresses because we want to send requests to
+				// pods before kube knows they are ready
+				for _, addr := range es.NotReadyAddresses {
+					// Prefer IP as we can avoid a DNS lookup this way.
+					ret = append(ret, net.JoinHostPort(addr.IP, portStr))
```

Comment:
If they are not ready, means that have not passed readiness checks yet. What is the point of using them in the list, since we can't probe them anyway (QP will fail the probe if it can't probe user container).
And when this transition happens we'll get the endpoints notification anyway.

Reply:
In our testing we saw added delay introduced by readyness probes even with our exec probing. Trying out this change locally, I see a significant reduction in the worst case times for cold-start.

Reply:
A bit more detail - although QP will fail until the application is ready, once QP detects a healthy application there is added delay while this information passes from kubelet through the k8s controlplane which we can eliminate.

Reply:
What I mean is that we won't see that anyway until 
1) endpoints send us a notification
2) our timed for loop triggers (which we know is a problem due to huge churn in GC, so we might need to relax it a bit, which defeats the purpose).

But sure, I don't really see a drawback here.